MovieApp/MovieApp/App/Repository/MovieRepository.swift

```diff
@@ -99,4 +159,18 @@ extension MovieRepository {
             completion(.failure(error))
         }
     }
+    
+    func updateCategoryMovies() {
+        let tempDict: [LocalCategory: [MovieRepoModel]] = categoryMovies.mapValues {
+            var movies: [MovieRepoModel] = []
+            for movie in $0 {
+                var x = movie
+                x.isFavorite = userDefaultsDataSource.favorites.contains(movie.identifier)
+                movies.append(x)
+            }
+            return movies
+        }
+        categoryMovies = tempDict
```

Comment:
Please try to avoid naming variables `tmpSomething` and `x` since it's hard to read.

Also you can write this in few lines:
```swift
categoryMovies = categoryMovies
    .mapValues { categoryMovies in
        categoryMovies
            .map { $0.copy(isFavorite: userDefaultsDataSource.favorites.contains($0.identifier)) }
    }
```

with that you need to create a `copy` function on the `MovieRepoModel`:
```swift
func copy(
    identifier: Int? = nil,
    title: String? = nil,
    backdropPath: String? = nil,
    posterPath: URL? = nil,
    overview: String? = nil,
    voteAverage: Double? = nil,
    voteCount: Double? = nil,
    releaseDate: String? = nil,
    genreIds: [Int]? = nil,
    budget: Int? = nil,
    runtime: Int? = nil,
    isFavorite: Bool? = nil
) -> MovieRepoModel {
    MovieRepoModel(
        identifier: identifier ?? self.identifier,
        title: title ?? self.title,
        backdropPath: backdropPath ?? self.backdropPath,
        posterPath: posterPath ?? self.posterPath,
        overview: overview ?? self.overview,
        voteAverage: voteAverage ?? self.voteAverage,
        voteCount: voteCount ?? self.voteCount,
        releaseDate: releaseDate ?? self.releaseDate,
        genreIds: genreIds ?? self.genreIds,
        budget: budget ?? self.budget,
        runtime: runtime ?? self.runtime,
        isFavorite: isFavorite ?? self.isFavorite)
}
```

I would suggest doing this because structures should be immutable and represent a snapshot of data at that moment (especially data models), any modification should create a new structure rather than editing an existing one. That will become very useful when dealing with Rx and multithreading since two threads can't access one object.

Reply:
[Copy func](https://github.com/lihter/Movie-App/pull/38/commits/cd9b26a2e35cb7418ace1f07855e235f95125719#diff-16ec28a5469846b8ad26272a1925c9f70fa4bf8572c2a83c4b764c9caa874313R54)

Not using `tempDict` anymore -> [here](https://github.com/lihter/Movie-App/pull/38/commits/cd9b26a2e35cb7418ace1f07855e235f95125719#diff-119fd471623eca33515c1b8fb62109b006cbeee1e1df1dc1a8e9a2e5ba5d4c27R167)